src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -3932,74 +3942,97 @@ public static SubnetGroups getInternodeErrorReportingExclusions()
     {
-        conf.track_warnings.coordinator_read_size.setAbortThresholdKb(threshold);
+        return conf.local_read_size_warn_threshold;
     }
 
-    public static long getLocalReadSizeWarnThresholdKb()
+    public static void setLocalReadSizeWarnThreshold(String threshold)
     {
-        return conf.track_warnings.local_read_size.getWarnThresholdKb();
+        DataStorageSpec value = parseDataStorageSpec(threshold);
+        logger.info("updated local_read_size_warn_threshold to {}", value);
+        conf.local_read_size_warn_threshold = value;
     }
 
-    public static void setLocalReadSizeWarnThresholdKb(long value)
+    public static DataStorageSpec getLocalReadSizeFailThreshold()
     {
-        conf.track_warnings.local_read_size.setWarnThresholdKb(value);
+        return conf.local_read_size_fail_threshold;
     }
 
-    public static long getLocalReadSizeAbortThresholdKb()
+    public static void setLocalReadSizeFailThreshold(String threshold)
     {
-        return conf.track_warnings.local_read_size.getAbortThresholdKb();
+        DataStorageSpec value = parseDataStorageSpec(threshold);
+        logger.info("updated local_read_size_fail_threshold to {}", value);
+        conf.local_read_size_fail_threshold = value;
     }
 
-    public static void setLocalReadSizeAbortThresholdKb(long value)
+    public static DataStorageSpec getRowIndexSizeWarnThreshold()
     {
-        conf.track_warnings.local_read_size.setAbortThresholdKb(value);
+        return conf.row_index_size_warn_threshold;
     }
 
-    public static int getRowIndexSizeWarnThresholdKiB()
+    public static void setRowIndexSizeWarnThreshold(String threshold)
     {
-        return conf.track_warnings.row_index_size.getWarnThresholdKb();
+        DataStorageSpec value = parseDataStorageSpec(threshold);
+        logger.info("updated row_index_size_warn_threshold to {}", value);
+        conf.row_index_size_warn_threshold = value;
     }
 
-    public static void setRowIndexSizeWarnThresholdKiB(int value)
+    public static DataStorageSpec getRowIndexSizeFailThreshold()
     {
-        conf.track_warnings.row_index_size.setWarnThresholdKb(value);
+        return conf.row_index_size_fail_threshold;
     }
 
-    public static int getRowIndexSizeAbortThresholdKiB()
+    public static void setRowIndexSizeFailThreshold(String threshold)
     {
-        return conf.track_warnings.row_index_size.getAbortThresholdKb();
+        DataStorageSpec value = parseDataStorageSpec(threshold);
+        logger.info("updated row_index_size_fail_threshold to {}", value);
+        conf.row_index_size_fail_threshold = value;
     }
 
-    public static void setRowIndexSizeAbortThresholdKiB(int value)
+    private static DataStorageSpec parseDataStorageSpec(String threshold)
     {
-        conf.track_warnings.row_index_size.setAbortThresholdKb(value);
+        return threshold == null
+               ? null
+               : new DataStorageSpec(threshold);
```

Comment:
What happens if someone passes something like `""` or `"      "`?

Reply:
both will fail due to `new DataStorageSpec` not being able to parse it; so we throw an exception.